pkg/auth/authproviders/openshift/internal/dexconnector/openshift.go

```diff
@@ -202,9 +204,23 @@ func validateEndpoint(endpoint string, tlsConfig *tls.Config) error {
 
 // LoginURL returns the URL to redirect the user to login with.
 func (c *openshiftConnector) LoginURL(_ connector.Scopes, callbackURL string, state string) (string, error) {
-	clonedConfig := *c.oauth2Config
-	clonedConfig.RedirectURL = callbackURL
-	return clonedConfig.AuthCodeURL(state, oauth2.SetAuthURLParam("redirect_uri", callbackURL)), nil
+	return c.oauth2Config.AuthCodeURL(state, oauth2.SetAuthURLParam("redirect_uri", callbackURL)), nil
+}
+
+// MakeRedirectURI constructs redirect URI from request info and the path.
+func MakeRedirectURI(ri *requestinfo.RequestInfo, path string) *url.URL {
+	scheme := "https"
+
+	// Allow HTTP only if the client did not use TLS and the host is localhost.
+	if !ri.ClientUsedTLS && netutil.IsLocalEndpoint(ri.Hostname) {
+		scheme = "http"
+	}
```

Comment:
```suggestion
	// Allow HTTP only if the client did not use TLS and the host is localhost.
	scheme := "https"
	if !ri.ClientUsedTLS && netutil.IsLocalEndpoint(ri.Hostname) {
		scheme = "http"
	}
```

I prefer grouping same things together for clarity. WDYT?

Reply:
Me too, that's why the comment about HTTP exception is close to the related checks. In your suggestion the comment looks belonging to the `scheme := "https"` line, which to me looks confusing.